src/js/mqtt.js

```diff
@@ -274,6 +274,11 @@ MqttClient.prototype.disconnect = function(err) {
   this._socket.end();
 };
 
+function qosEqual0(qos) {
+  if (!qos || Number.isNaN(qos)) return false;
```

Comment:
What the case that qos is NaN?

Reply:
when the `options` argument is undefined

Reply:
undefined has been covered at `!qos`